app/models/settings.rb

```diff
@@ -98,13 +97,25 @@ def spam_protection_level
     level == :auto ? :captcha : level
   end
 
-  protected
-
   delegate :provider_id_for_audits, :to => :account, :allow_nil => true
 
   private
 
   def not_custom_account_plans
     @not_custom_account_plans ||= provider.account_plans.not_custom
   end
+
+  def default_account_plan
+    provider.account_plans.default || not_custom_account_plans.first!
+  end
+
+  def update_approval_required(attrs)
+    value = attrs.delete(:account_approval_required).presence || false
+    default_account_plan.update_attribute(:approval_required, value)
+  end
+
+  # Remove attributes with empty strings and nil for non-null columns
+  def sanitize_attributes(attrs)
+    attrs.reject { |key, value| self.class.non_null_columns_names.include?(key.to_s) && value.to_s.empty? }
```

Comment:
```suggestion
    attrs.reject { |key, value| self.class.non_null_columns_names.include?(key.to_s) && value.blank? }
```

Is it not better to use `blank?` here?, it never raises.

![rails_blank cheatsheet](https://github.com/3scale/porta/assets/17052241/0fe0755f-216f-456d-af80-ac7cea7d9f9a)

Reply:
I guess we don't want to treat `false` as non-blank in boolean columns.

Reply:
`.blank?` would reject `false` value, and I don't want that because this is a valid one.

It is really aimed for `""` and `nil` because these are the ones that cause troubles with non-null columns. So, we could also just check `["",nil].include?(value)` instead. Do you think that's better?